lib/common.sh

```diff
@@ -104,15 +104,16 @@ clone_repo() {
   if [[ -d "${REPO_PATH}" ]] && [[ "${FORCE_REPO_UPDATE}" = "true" ]]; then
     rm -rf "${REPO_PATH}"
   fi
-  if [[ ! -d "${REPO_PATH}" ]] ; then
+  if [[ ! -d "${REPO_PATH}" ]]; then
     pushd "${M3PATH}" || exit
-    git clone "${REPO_URL}" "${REPO_PATH}"
-    popd || exit
-    pushd "${REPO_PATH}" || exit
-    git checkout "${REPO_BRANCH}"
-    git checkout "${REPO_COMMIT}"
-    git pull -r || true
+    git clone --depth 1 --branch "${REPO_BRANCH}" "${REPO_URL}" "${REPO_PATH}"
     popd || exit
+    if [[ "${REPO_COMMIT}" != "HEAD" ]]; then
+        pushd "${REPO_PATH}" || exit
+        git checkout "${REPO_COMMIT}"
```

Comment:
This will fail with `--depth 1`. With depth 1, it does not know any other commit than HEAD.

```
git clone --depth 1 --branch main https://github.com/metal3-io/project-infra project-infra-1
# it doesn't know HEAD~2 for example
❯ git checkout HEAD~2
error: pathspec 'HEAD~2' did not match any file(s) known to git
# it doesn't know a sha that is couple commits in history
❯ git checkout 7c8282e6f015246068044cd6b17d66eb4ab2a7db
fatal: reference is not a tree: 7c8282e6f015246068044cd6b17d66eb4ab2a7db
```

/hold

Reply:
thanks, now it should be better